### PR TITLE
Fix bug with plot close/open cycle and permanent markers

### PR DESCRIPTION
**Pull request by ericpre**
First follow up of #3148 - hopefully, there will *not* be many! 😅

### CODE REVIEW

File: hyperspy/drawing/markers.py [hunk 1]
```diff
@@ -657,7 +657,7 @@ def get_current_kwargs(self, only_variable_length=False):
         """
         current_keys = {}
         if self._is_iterating:
-            indices = self._axes_manager.indices[::-1]
```
Comment (CSSFrancis) [inline hyperspy/drawing/markers.py:660]: This is currently causing the `from_signal` method to silently fail.  You need to reverse the indexes to follow the array--> plotting specification.

**Review by CSSFrancis — CHANGES_REQUESTED** [review]
The reversing needs to happen in order to account for the reverse ordering with respect to array/plotting. 

At the very least we should maybe make this addition to the `test_find_peaks` function.


```python 
    def test_find_peaks(self):
        from skimage.draw import disk
        from skimage.morphology import disk as disk2
        rr, cc = disk(
            center=(10, 8),
            radius=4,
        )
        img = np.zeros((2,2, 20, 20))
        img[:,:, rr, cc] = 1
        img[1,0] = 0
        s = Signal2D(img)
        s.axes_manager.signal_axes[0].scale = 1.5
        s.axes_manager.signal_axes[1].scale = 2
        s.axes_manager.signal_axes[0].offset = -1
        s.axes_manager.signal_axes[1].offset = -1
        pks = s.find_peaks(
            interactive=False,
            method="template_matching",
            template=disk2(4),
        )
        col = Points.from_signal(
            pks, sizes=(10,), signal_axes=s.axes_manager.signal_axes
        )
        s.add_marker(col)
        np.testing.assert_array_equal(col.get_current_kwargs()["offsets"], [[11, 19]])
        a = s.axes_manager.indices = (0,1) 
        np.testing.assert_array_equal(col.get_current_kwargs()["offsets"], [[np.nan, np.nan]])
```

**Comment by ericpre** [comment]
Thank you @CSSFrancis, the example that you gave here illustrate clearly the issue! I used an another approach: transpose the data array (only when ragged, because static marker don't have navigation dimension). I added tests to check the navigation shape for all relevant.

**Comment by CSSFrancis** [comment]
@ericpre ahh perfect!  I was apparently only thinking about the ragged case when I wrote that 😅 The added test is great!

This looks good as long as the CI passes.

**Review by CSSFrancis — APPROVED** [review]

**Pull request merged**